src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -982,12 +990,18 @@ private void AddDependencies(Dictionary<String, Tuple<PackageArchiveReader, Pack
                     }
 
                     var dependency = new PackageDependency(packageReference.PackageIdentity.Id, range);
+                    var packageName = $"{packageReference.PackageIdentity.Id}.{packageReference.PackageIdentity.Version}";
+
+                    var stream = finder.GetNupkgStreamAsync(packageReference.PackageIdentity.Id, packageReference.PackageIdentity.Version, CancellationToken.None).Result;
 
-                    var path = Path.Combine(_project.DirectoryPath, PackagesFolder, $"{packageReference.PackageIdentity.Id}.{packageReference.PackageIdentity.Version}.nupkg");
-                    if (File.Exists(path))
+                    if (stream != null)
+                    {
+                        var reader = new PackageArchiveReader(stream);
+                        packagesAndDependencies.Add(packageReference.PackageIdentity.Id, new Tuple<PackageReaderBase, PackageDependency>(reader, dependency));
+                    }
+                    else
                     {
-                        PackageArchiveReader reader = new PackageArchiveReader(path);
-                        packagesAndDependencies.Add(packageReference.PackageIdentity.Id, new Tuple<PackageArchiveReader, PackageDependency>(reader, dependency));
+                        throw new CommandLineException(NuGetResources.UnableToFindBuildOutput, $"{packageName}.nupkg");
```

Comment:
Is this the most appropriate error message? This is more of a restore output that build, but maybe this is standard to the rest of the product.